File: Pi_Dev_Cloud-Hilmi-back/src/main/java/com/esprit/elearningback/entity/Service.java
package com.esprit.elearningback.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
@Table(name = "service")
public class Service {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long serviceId;

    private String serviceName;

    @Column(length = 5000)
    private String serviceDescription;

    private String imageUrl;

    @JsonIgnore
    @OneToMany(mappedBy = "service", cascade = CascadeType.ALL)
    private List<Ressources> ressources;

    @ManyToOne
    @JoinColumn(name = "subject_id" , nullable= true)
    private Subject subject;

    @JsonIgnore
    @OneToMany(mappedBy = "service", cascade = CascadeType.ALL)
    private List<Rating> Ratings;

}

File: Frontend/src/app/model/service.ts
import { Subject } from './subject'; // Assuming you have a Subject model

export class Service {
    serviceid: number;
    service_name: string;
    service_description: string;
    subject: Subject;
  
    constructor(serviceid: number, service_name: string, service_description: string, subject: Subject) {
      this.serviceid = serviceid;
      this.service_name = service_name;
      this.service_description = service_description;
      this.subject = subject;
    }
  }